### PR TITLE
203 202 create song hints

### PR DESCRIPTION
**Pull request by Flamantrose321**

### CODE REVIEW

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameView.kt [hunk 1]
```diff
@@ -72,6 +73,9 @@ interface GameView: Timer.DeadlineHandler {
      */
     fun updateLyrics(lyrics : String)
 
+
+    fun addHint(songNameHint: SongNameHint)
+
```
Comment (nathanduchesne) [inline app/src/main/java/com/github/fribourgsdp/radio/game/GameView.kt:78]: Please add doc for this method (as in this whole interface).

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt [hunk 2]
```diff
@@ -306,6 +322,29 @@ open class GameActivity : AppCompatActivity(), GameView, Timer.Listener {
...
+    fun updateHint(timeInSeconds: Int){
+        if(withHint && hintTextView.visibility == View.VISIBLE) {
+            val limit = ceil(gameDuration/ (ceil(songNameHint.length()/2.0)+1))
+            if(timeInSeconds - lastTime >= limit) {
+                songNameHint.addALetter()
+                hintTextView.text = this.songNameHint.toString()
+                lastTime = timeInSeconds
+            }
+        }
+    }
```
Comment (nathanduchesne) [inline app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt:344]: This algorithm makes it such that at the end of the timer, half of the letters in the song's title have been displayed. Is this maybe too much? I think we should ask the group to see if 1/3 of the total title is enough.

Comment (Flamantrose321) [reply app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt:344]: I think half the letters are good as they appear one after the other but if others think like you I will change.

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt [hunk 3]
```diff
@@ -128,6 +134,10 @@ open class GameActivity : AppCompatActivity(), GameView, Timer.Listener {
             text = songName
             visibility = View.VISIBLE
         }
+
+        if(withHint) {
+            hintTextView.visibility = View.GONE
+        }
```
[the remark this answers is not in the document]
Comment (Flamantrose321) [reply app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt:140]: Yes, that's right. And addhint is called for every new song you guess.

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt (hunk 2 above)
Comment (BartLucien) [reply app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt:344]: We tested (with @nathanduchesne) on scribble and at the end there were only 2 letters on small words. I think 1/2 is a bit too much, as one can guess the title just by building the words from the letters that have appeared. 1/3 or 1/4 letters seems a betters choice. Changing the value shouldn't be too difficult if we have to change it anyway.

File: app/src/main/java/com/github/fribourgsdp/radio/util/SongNameHint.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+
+    fun stripAccents(s : String): String {
+        var s2 = Normalizer.normalize(s, Normalizer.Form.NFD)
+        s2 = s2.replace("[\\p{InCombiningDiacriticalMarks}]".toRegex(), "")
+        return s2
+    }
+
+    fun replaceWithLine(s : String):String{
+        val re = Regex("[A-Za-z0-9]")
```
[the remark this answers is not in the document]
Comment (Flamantrose321) [reply app/src/main/java/com/github/fribourgsdp/radio/util/SongNameHint.kt:40]: Oui c'est cela. Toutes les lettres ( a, b, c A, b c, ç, é, ö, â, .....) et chiffres sont remplacé par des _

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt (hunk 2 above)
Comment (nathanduchesne) [reply app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt]: Thanks for changing it to 1/3. Everything is fine for me now.

**Review by sim0mo — APPROVED** [review]
Ok with this feature ! It makes the game very playable

**Pull request merged**